src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageViewModel.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Common.Core;
+using Microsoft.Common.Wpf;
+using Microsoft.R.Components.PackageManager.Model;
+using Microsoft.R.Components.PackageManager.ViewModel;
+
+namespace Microsoft.R.Components.PackageManager.Implementation.ViewModel {
+    internal class RPackageViewModel : BindableBase, IRPackageViewModel {
+        private bool _hasDetails;
+        private bool _isSelected;
+        private bool _isInstalled;
+        private string _title;
+        private string _description;
+        private string _authors;
+        private string _installedVersion;
+        private string _latestVersion;
+        private ICollection<string> _urls;
+        private string _libraryPath;
+        private string _built;
+
+        public static RPackageViewModel CreateAvailable(RPackage package) {
+            return new RPackageViewModel(package.Package) {
+                LatestVersion = package.Version,
+                Depends = package.Depends,
+                Imports = package.Imports,
+                Suggests = package.Suggests,
+                License = package.License,
+                NeedsCompilation = package.NeedsCompilation != null && !package.NeedsCompilation.EqualsIgnoreCase("no"),
+                Repository = package.Repository,
+            };
+        }
+
+        public static RPackageViewModel CreateInstalled(RPackage package) {
+            return new RPackageViewModel(package.Package) {
+                Title = package.Title,
+                Authors = package.Author,
+                License = package.License,
+                Urls = package.URL?.Split(new[] { "," }, StringSplitOptions.RemoveEmptyEntries)
```

Comment:
You're right. Weird that they did this. I would have expected to have to keep chaining `?.`, unless I wanted one of the calls to actually happen for `null` (e.g. what if I have an extension method that handles nulls in a special manner?).